### PR TITLE
feat: Walk

### PR DESCRIPTION
**Pull request by Prakkkmak**

### CODE REVIEW

File: code/Player/Controllers/HumanController.cs [hunk 1]
```diff
@@ -75,7 +91,7 @@ public override void Simulate()
                 else
                 {
                     ApplyFriction(GroundSurface.Friction * SurfaceFriction);
-                    Walk();
+                    Move();
```
Comment (Nowp) [inline code/Player/Controllers/HumanController.cs:94]: That is not only move, this is when we're moving on ground. Maybe call this function MoveGround then ? Just "Move" is too general

File: code/Player/Controllers/HumanController.cs [hunk 2]
```diff
@@ -168,28 +184,20 @@ private void ProcessInputs()
                 WishVelocity = EyeRot * WishVelocity.ClampLength(1);
             }
 
-            IsSprinting = Input.Down(InputButton.Run);
+            if (Input.Down(InputButton.Run)) state = MovementState.Sprint;
+            else if (Input.Down(InputButton.Walk)) state = MovementState.Walk;
+            else if(WishVelocity.Length > 0) state = MovementState.Run;
```
Comment (Nowp) [inline code/Player/Controllers/HumanController.cs:189]:
```suggestion
            else if (WishVelocity.Length > 0) state = MovementState.Run;
```

File: code/Player/Controllers/HumanController.cs (hunk 1 above)
Comment (Prakkkmak) [reply code/Player/Controllers/HumanController.cs:94]: yeah but the method moves you and don't care if youre on ground

Comment (Nowp) [reply code/Player/Controllers/HumanController.cs:94]: It applies the logic of ground, air move is not the same, swim move would not be the same and also no gravity live.
Depending on the state of the player, different logic is applied to make him move

Comment (Prakkkmak) [reply code/Player/Controllers/HumanController.cs]: Done do you prefer ?

**Review by Nowp — APPROVED** [review]
I would call the method "GroundMove" or at least remove the "accelerate" but besides that it's ok

**Pull request merged**